src/terminfo/parser/compiled.rs

```diff
@@ -157,8 +158,22 @@ pub static stringnames: &'static[&'static str] = &[ "cbt", "_", "cr", "csr", "tb
     "OTG3", "OTG1", "OTG4", "OTGR", "OTGL", "OTGU", "OTGD", "OTGH", "OTGV", "OTGC", "meml", "memu",
     "box1"];
 
+fn read_harder<R: io::Read>(reader: &mut R, buffer: &mut [u8]) -> io::Result<usize> {
```

Comment:
Sounds reasonable. However, `read_to_end` ignores `ErrorKind::Interrupted` (as does `byteorder` as of a few days ago... https://github.com/BurntSushi/byteorder/pull/16).